galpy/potential/potential_c_ext/MovingObjectPotential.c

```diff
@@ -0,0 +1,158 @@
+#include <math.h>
+#include <galpy_potentials.h>
+
+#include <gsl/gsl_errno.h>
+#include <gsl/gsl_spline.h>
+// MovingObjectPotential
+// 5 fixed arguments: amp, t0, tf, nsteps, ndim
+// Followed by <nsteps> time values and <ndim>*<nsteps> coordinates
+void constrain_range(double * d) {
+  // Constrains index to be within interpolation range
+  if (*d < 0) *d = 0.0;
+  if (*d > 1) *d = 1.0;
+}double MovingObjectPotentialRforce(double R,double z, double phi,
+				      double t,
+				      struct potentialArg * potentialArgs){
+  double * args= potentialArgs->args;
+  //Get args
+  double amp= *args;
+  double t0= *(args+1);
+  double tf= *(args+2);
+  int n_steps= (int) *(args+3);
```

Comment:
@turnergarrow: I don't think ``n_steps``, ``n_dim``, or ``o`` is used in this function (or in any of the other functions). See my other comment about not copying these to ``args``.